### ISSUE TITLE
SecondaryPermissions is returning string in array instead of array

### ISSUE THREAD

**Issue by vanvantsyan** [cartalyst/sentinel#441]
Sentinel::hasAccess($some_route)

For just Users roles it works fine , but if user has some Role , when getting the Role permissions from DB , it returns a string (like : "{"users.create":true,"users.delete":true,}") instead of array . It causes an error:

Type error: Argument 2 passed to Cartalyst\Sentinel\Permissions\StandardPermissions::preparePermissions() must be of the type array, string given

I've found this solution , to add:
 "$permissions = (array)json_decode($permissions);" 
for secondary permission to make it array , but that's not a permanent and right way to fix error in vendor.

protected function createPreparedPermissions()
    {
        $prepared = [];

        if (! empty($this->secondaryPermissions)) {
            foreach ($this->secondaryPermissions as $permissions) {
                $permissions = (array)json_decode($permissions);
                $this->preparePermissions($prepared, $permissions);
            }
        }
        if (! empty($this->permissions)) {
            $permissions = [];

            $this->preparePermissions($permissions, $this->permissions);

            $prepared = array_merge($prepared, $permissions);
        }

        return $prepared;
    }

Please , help me to solve this.

**Comment by brunogaspar** [comment]
You most likely tampered with the data manually or incorrectly saved using the query builder or something.

I'm saying this because `{"users.create":true,"users.delete":true,}` is not valid JSON.

**Comment by vanvantsyan** [comment]
@brunogaspar Thank you so much for quick response.
It will be a great help if you advice a way to solve this. Thank you.
This one is the real value role permissions.
{"users.create":true,"users.store":true,"users.destroy":true,"users.view":true,"users.update":true}
The same value for user permissions.

In PermissionTrait.php
    public function __construct(array $permissions = null, array $secondaryPermissions = null)
    {
        if (isset($permissions)) {
           dd($permissions);
            $this->permissions = $permissions;
        }
        if (isset($secondaryPermissions)) {
            dd($secondaryPermissions);
            $this->secondaryPermissions = $secondaryPermissions;
        }
    }
$permission (first) dump returns :
array:5 [▼
  "users.create" => true
  "users.store" => true
  "users.destroy" => true
  "users.view" => true
  "users.update" => true
]
$secondaryPermissions (second) dump returns:
array:1 [▼
  0 => "{"users.create":true,"users.store":true,"users.destroy":true,"users.view":true,"users.update":true}"
]

In case migrations run the same way for both and the same values are in role and users for permission.

Please help me to find the reason , it's too important.

**Comment by brunogaspar** [comment]
What's the code you're using that causes this? I'm not being able to reproduce.

**Comment by mwolfeconsultant** [comment]
Hello - I reviewed Van's code and found the issue. You were correct in that it was on our end and I appreciate your input.

I'll go ahead and note here what I found as to potentially assist anyone else who may run into this issue in the future. The problem was our User model referenced a different Role model, and this didn't allow Sentinel's default Role model to be used. 

As Brungogaspar has been saying on this issue and #242 - the issue was up the stack in how the data was being pulled, not how Sentinel was accessing or using it.
